web_widget_float_highlight/static/src/js/web_highlight.js

```diff
@@ -0,0 +1,72 @@
+odoo.define("web.float.highlight", function (require) {
+    "use strict";
+    var form_widgets = require("web.form_widgets");
+
+    form_widgets.FieldFloat.include({
+
+        init: function () {
+            this._super.apply(this, arguments);
+            var options = {
+                lower_threshold: (this.options && this.options.lower_threshold) || 0,
+                upper_threshold: (this.options && this.options.upper_threshold) || 0,
+                lower_bg_color: (this.options && this.options.lower_bg_color) || "red",
+                middle_bg_color: (this.options && this.options.middle_bg_color) || "white",
+                upper_bg_color: (this.options && this.options.upper_bg_color) || "green",
+                lower_txt_color: (this.options && this.options.lower_txt_color) || "white",
+                middle_txt_color: (this.options && this.options.middle_txt_color) || "#666666",
+                upper_txt_color: (this.options && this.options.upper_txt_color) || "white",
+                always_work: (this.options && this.options.always_work) || false,
+                highlight: (this.options && this.options.highlight) || false
+            };
+            this.lower_threshold = options.lower_threshold;
```

Comment:
Are those declarations necessary? Why don't you use this.options throughout the widget?